metabase/metabase
# Add popover/embed modal display logic to the dashboard embed button

Epic: #35961 

### Description

For the embed button, if:
* User is an admin and public sharing is enabled, then the user will see `Create a public link` and the `Embed` option
* User is an admin and public sharing is disabled, the user will see the `Static Embed` modal
* User is an admin and public sharing & embedding is disabled, the user will still see the embedding modal
* User is a non-admin and a public link has been created for the dashboard, they will see the Embed button with the label of `Sharing`
* User is a non-admin and a public link has not been created for the dashboard, they will not see the button at all

**Note: the actual dropdown for creating the public link has not been made yet. This is the first step in Milestone 1, in which the next task will be to create/remove the link.**

### How to verify

_**As an admin**_
* Public Sharing enabled:
	* Should see a dropdown with `Create a public link` and `Embed` as options, where clicking `Embed` should take the user to the modal, and `Create a public link` does nothing
* Public sharing disabled:
	* Should immediately be directed to the Embed modal


_**As a non-admin**_
* Public sharing enabled & link is created:
	* you should see the Embed button with the label of `Sharing`. The popover isn't there yet.
* Public sharing disabled or link hasn't been created:
	* you shouldn't see an `Embed` button 


### Demo

https://github.com/metabase/metabase/assets/25306947/affeef71-ebfa-4f40-8c40-df06e0177bce



### Checklist
